src/modules/commander/Commander.cpp

```diff
@@ -3994,51 +3996,69 @@ void Commander::estimator_check()
 		if (_estimator_selector_status_sub.copy(&estimator_selector_status)) {
 			if (estimator_selector_status.primary_instance != _estimator_status_sub.get_instance()) {
 				_estimator_status_sub.ChangeInstance(estimator_selector_status.primary_instance);
+				_estimator_status_flags_sub.ChangeInstance(estimator_selector_status.primary_instance);
 			}
 		}
 	}
 
-	if (_estimator_status_sub.update()) {
-		const estimator_status_s &estimator_status = _estimator_status_sub.get();
+	if (_estimator_status_flags_sub.update()) {
+		const estimator_status_flags_s &estimator_status_flags = _estimator_status_flags_sub.get();
 
-		// Check for a magnetomer fault and notify the user
-		const bool mag_fault = (estimator_status.control_mode_flags & (1 << estimator_status_s::CS_MAG_FAULT));
-		const bool gnss_heading_fault = (estimator_status.control_mode_flags & (1 << estimator_status_s::CS_GPS_YAW_FAULT));
+		// TODO: also consider estimator_status_flags.cs_inertial_dead_reckoning
+		_status_flags.dead_reckoning = estimator_status_flags.cs_wind_dead_reckoning;
```

Comment:
should we have `dead_reckoning = cs_wind_dead_reckoning || cs_inertial_dead_reckoning` ?

Reply:
That's what I had initially, but removed it in https://github.com/PX4/PX4-Autopilot/pull/18787 where it was just an unnecessary switch from MISSION -> RTL (very briefly) -> DESCEND (failsafe) for the few seconds of inertial dead reckoning.

Reply:
Fixed for now. We'll need to do something different for the state machine (https://github.com/PX4/PX4-Autopilot/pull/18787).